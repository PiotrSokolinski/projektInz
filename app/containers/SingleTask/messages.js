/*
 * SingleTask Messages
 *
 * This contains all the text for the SingleTask container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.SingleTask'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the SingleTask container!',
  },
  taskName: {
    id: `${scope}.taskName`,
    defaultMessage: 'Name: {name}',
  },
  taskStatus: {
    id: `${scope}.taskStatus`,
    defaultMessage: 'Status:',
  },
  authorName: {
    id: `${scope}.authorName`,
    defaultMessage: 'Author: {firstName} {lastName}',
  },
  assigneeName: {
    id: `${scope}.authorName`,
    defaultMessage: 'Assignee: {firstName} {lastName}',
  },
  showMore: {
    id: `${scope}.showMore`,
    defaultMessage: 'Show more...',
  },
  showLess: {
    id: `${scope}.showLess`,
    defaultMessage: 'Show less...',
  },
  description: {
    id: `${scope}.description`,
    defaultMessage: 'Description',
  },
  taskPriority: {
    id: `${scope}.priority`,
    defaultMessage: 'Priority:',
  },
  createdAt: {
    id: `${scope}.createdAt`,
    defaultMessage: 'Created at:',
  },
})
