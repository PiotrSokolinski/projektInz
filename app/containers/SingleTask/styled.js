import styled from 'styled-components'
import { ArrowDownThick } from 'styled-icons/typicons/ArrowDownThick'
import { ArrowUp } from 'styled-icons/feather/ArrowUp'

import { Colors } from 'themes'

export const ArrowDown = styled(ArrowDownThick)`
  ${props => props.isShowMore && 'transform: rotate(-180deg)'};
`

const getArrowPosition = props => {
  switch (props.priority) {
    case 'High':
      return ''
    case 'Medium':
      return 'rotate(-90deg)'
    case 'Low':
      return 'rotate(-180deg)'
    default:
      return ''
  }
}

const getArrowColor = props => {
  switch (props.priority) {
    case 'High':
      return 'red'
    case 'Medium':
      return 'orange'
    case 'Low':
      return 'green'
    default:
      return 'red'
  }
}

export const Arrow = styled(ArrowUp)`
  margin-left: 8px;
  transform: ${props => getArrowPosition(props)};
  color: ${props => getArrowColor(props)};
`

export const SingleTaskContainer = styled.div`
  margin: 0 20px;
  border: 1px solid ${Colors.edward};
  margin-top: 20px;
  border-radius: 20px;
`

export const InfoContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 20px;
  padding: 0 25px;
  margin-top: 5px;
`

export const TaskName = styled.div`
  cursor: pointer;
`

export const Status = styled.div`
  display: flex;
  align-items: center;
`

const getDotBackgroundColor = props => {
  switch (props.status) {
    case 'To Do':
      return 'red'
    case 'In Progress':
      return 'orange'
    case 'Done':
      return 'green'
    default:
      return 'red'
  }
}

export const Dot = styled.div`
  background-color: ${props => getDotBackgroundColor(props)};
  border-radius: 50%;
  height: 8px;
  margin-right: 8px;
  margin-left: 8px;
  width: 8px;
`

export const PersonContainer = styled.div`
  display: flex;
  align-items: center;
`

export const FullNameContainer = styled.div`
  margin-right: 10px;
`

export const ShowMoreContainer = styled.div`
  display: flex;
  justify-content: center;
`

export const ShowMoreButton = styled.div`
  display: flex;
  cursor: pointer;
  align-items: center;
  height: 20px;
`

export const DescriptionContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: 20px;
  padding: 0 25px;
  margin-top: 5px;
`

export const Description = styled.div``

export const Priority = styled.div``

export const PriorityContainer = styled.div`
  display: flex;
  align-items: center;
`

export const CreatedAtContainer = styled.div`
  display: flex;
  align-items: center;
`

export const CreatedAt = styled.div`
  margin-left: 5px;
`
