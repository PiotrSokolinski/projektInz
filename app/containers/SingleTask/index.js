/**
 *
 * SingleTask
 *
 */

import React, { useState } from 'react'
import dayjs from 'dayjs'
import PropTypes from 'prop-types'
import get from 'lodash/get'
import { FormattedMessage } from 'react-intl'

import UserAvatar from 'components/UserAvatar'
import Modal from 'components/Modal'
import EditTask from 'containers/EditTask'

import messages from './messages'
import * as Styled from './styled'

const SingleTask = ({ task }) => {
  const [isShowMore, setIsShowMore] = useState(false)
  const [isModalTaskVisible, setIsModalTaskVisible] = useState(false)
  const openTaskModal = () => setIsModalTaskVisible(true)
  const closeTaskModal = () => setIsModalTaskVisible(false)
  const toggleShowMore = () => setIsShowMore(!isShowMore)

  return (
    <Styled.SingleTaskContainer>
      <Styled.InfoContainer>
        <Styled.TaskName onClick={openTaskModal}>
          <FormattedMessage {...messages.taskName} values={{ name: get(task, 'name', '') }} />
        </Styled.TaskName>
        <Styled.Status>
          <FormattedMessage {...messages.taskStatus} />
          <Styled.Dot status={task.status} />
          {get(task, 'status', '')}
        </Styled.Status>
      </Styled.InfoContainer>
      <Styled.InfoContainer>
        <Styled.PersonContainer>
          <Styled.FullNameContainer>
            <FormattedMessage
              {...messages.authorName}
              values={{ firstName: get(task, 'author.firstName', ''), lastName: get(task, 'author.lastName', '') }}
            />
          </Styled.FullNameContainer>
          <UserAvatar image={task.author.avatarUrl} size="tiny" />
        </Styled.PersonContainer>
        <Styled.PersonContainer>
          <Styled.FullNameContainer>
            <FormattedMessage
              {...messages.assigneeName}
              values={{ firstName: get(task, 'assignee.firstName', ''), lastName: get(task, 'assignee.lastName', '') }}
            />
          </Styled.FullNameContainer>
          <UserAvatar image={task.assignee.avatarUrl} size="tiny" />
        </Styled.PersonContainer>
      </Styled.InfoContainer>
      {isShowMore && (
        <React.Fragment>
          <Styled.DescriptionContainer>
            <FormattedMessage {...messages.description} />
            <Styled.Description>{get(task, 'description', '')}</Styled.Description>
          </Styled.DescriptionContainer>
          <Styled.InfoContainer>
            <Styled.PriorityContainer>
              <FormattedMessage {...messages.taskPriority} />
              <Styled.Arrow priority={task.priority} size="22" />
              <Styled.Priority>{get(task, 'priority', '')}</Styled.Priority>
            </Styled.PriorityContainer>
            <Styled.CreatedAtContainer>
              <FormattedMessage {...messages.createdAt} />
              <Styled.CreatedAt>{dayjs(get(task, 'createdAt', '')).format('DD/MM/YYYY HH:mm:ss')}</Styled.CreatedAt>
            </Styled.CreatedAtContainer>
          </Styled.InfoContainer>
        </React.Fragment>
      )}
      <Styled.ShowMoreContainer>
        <Styled.ShowMoreButton onClick={toggleShowMore}>
          <FormattedMessage {...messages[isShowMore ? 'showLess' : 'showMore']} />
          <Styled.ArrowDown size="22" isShowMore={isShowMore} />
        </Styled.ShowMoreButton>
      </Styled.ShowMoreContainer>
      <Modal title="Details" visible={isModalTaskVisible} onClose={closeTaskModal}>
        <EditTask taskId={task.id} onClose={closeTaskModal} />
      </Modal>
    </Styled.SingleTaskContainer>
  )
}

SingleTask.propTypes = {
  task: PropTypes.object,
}

export default SingleTask
