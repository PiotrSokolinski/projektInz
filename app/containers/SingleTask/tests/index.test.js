/**
 *
 * Tests for SingleTask
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import dayjs from 'dayjs'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import SingleTask from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<SingleTask />', () => {
  const task = {
    taskName: 'Task Name',
    taskDescription: `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.`,
    author: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
    createdAt: dayjs().format('DD/MM/YYYY HH:mm:ss'),
    priority: 'High',
    status: 'Done',
    assignee: {
      firstName: 'John',
      lastName: 'Doe',
      avatarUrl: '',
    },
  }
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <SingleTask task={task} />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <SingleTask task={task} />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
