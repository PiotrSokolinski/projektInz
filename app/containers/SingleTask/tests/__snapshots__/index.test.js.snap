// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleTask /> Should render and match the snapshot 1`] = `
<div
  class="styled__SingleTaskContainer-phousv-2 sZbZA"
>
  <div
    class="styled__InfoContainer-phousv-3 eHHDhm"
  >
    <div
      class="styled__TaskName-phousv-4 bupyFw"
    >
      <span>
        Name: Task Name
      </span>
    </div>
    <div
      class="styled__Status-phousv-5 dShLoI"
    >
      <span>
        Status:
      </span>
      <div
        class="styled__Dot-phousv-6 hLLxni"
      />
      Done
    </div>
  </div>
  <div
    class="styled__InfoContainer-phousv-3 eHHDhm"
  >
    <div
      class="styled__PersonContainer-phousv-7 cVJQRW"
    >
      <div
        class="styled__FullNameContainer-phousv-8 jvNBev"
      >
        <span>
          Author: John Doe
        </span>
      </div>
      <div
        class="tiny styled__Container-vinm4t-0 gDIJTt"
      >
        <img
          alt="avatar"
          class="avatar-image"
          src="IMAGE_MOCK"
        />
      </div>
    </div>
    <div
      class="styled__PersonContainer-phousv-7 cVJQRW"
    >
      <div
        class="styled__FullNameContainer-phousv-8 jvNBev"
      >
        <span>
          Assignee: John Doe
        </span>
      </div>
      <div
        class="tiny styled__Container-vinm4t-0 gDIJTt"
      >
        <img
          alt="avatar"
          class="avatar-image"
          src="IMAGE_MOCK"
        />
      </div>
    </div>
  </div>
  <div
    class="styled__ShowMoreContainer-phousv-9 flzkyG"
  >
    <div
      class="styled__ShowMoreButton-phousv-10 fQaDDG"
    >
      <span>
        Show more...
      </span>
      <svg
        aria-hidden="true"
        class="styled__ArrowDown-phousv-0 fiXBOw StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.414 10.656a2 2 0 00-2.828 0L14 12.242V5a2 2 0 00-4 0v7.242l-1.586-1.586a2 2 0 10-2.828 2.828L12 19.898l6.414-6.414a2 2 0 000-2.828z"
        />
      </svg>
    </div>
  </div>
</div>
`;
